### ISSUE TITLE
[Support Request] MariaDB Galera rolling Upgrades based on 10.7 to 10.8 images fail

### ISSUE THREAD

**Issue by jerrac** [bitnami/containers#8115]
### Name and Version

bitnami/mariadb-galera:10.8

### What steps will reproduce the bug?

* Initialize a 10.7 cluster on Docker Swarm. (I know Docker Swarm has no official support. I think this issue is not Swarm related.)
* Switch the image to 10.8 and redeploy.

### What is the expected behavior?

Each of the 10.7 replicas/containers should be replaced by a new 10.8 replica/container.

### What do you see instead?

One of the old containers is stopped, and a new 10.8 container is started. It seems to join the cluster, but then something causes it to fail.

### Additional information

A gist with all my files and logs is here: https://gist.github.com/jerrac/6529bfdbd790f1aae7d8de31a999726d

Long story short, I'm wrapping the Bitnami entrypoint script to get around some of Docker Swarm's issues. Initializing the cluster works. But upgrading fails. 

I'm not even sure which log messages are relevant, or what I might search for to find the answers. That's why I put everything into that gist.

My goal with this project is to just prove that we can run a containerized Galera cluster. So I haven't had time to actually dig in and learn Galera like I need to...

Anyway, would someone with more Galera experience mind skimming my gist and seeing if anything pops out at them?

Thanks in advance!

**Comment by javsalgar** [comment]
Hi,

Were you able to reproduce the same issue in a Kubernetes cluster? Just to confirm that the issue is not Swarm related.

**Comment by jerrac** [comment]
@javsalgar Er, I'll see if I can figure how to test with Kube, but the whole point of Swarm for my team is to avoid Kube... (Long story short, Kube is more complicated than my 2 person team can handle, or really need. Swarm isn't that complicated.)

**Comment by jerrac** [comment]
I did try to test on Kube. Didn't get far. The Bitnami Helm chart didn't work on Minikube for me and I wasn't able to troubleshoot it enough to figure out if it was me or the chart. And when I looked into adjusting my wrapped version of the Galera image, I got stuck on figuring out how to find the replica siblings. 

Specifically, I'm not sure what Kube's equivalent to `tasks.< service name >` is. 

At this point I'm largely out of time to get this working. So I'll likely end up just managing Galera outside of whatever orchestration tool I use. Either a straight vm based cluster, or maybe a set of standalone docker instances. 

Anyway, if anyone sees anything in my gist that could point me in the right direction, I'd still like to pursue this at least a bit. Otherwise I'll just let this go stale and auto close.

**Comment by javsalgar** [comment]
Hi,

Could you share with us what the error was when deploying the chart?

**Comment by jerrac** [comment]
```yaml
image:
  tag: 10.7
rootUser:
  password: password
db:
  user: lanegalera2
  password: password
  name: lanegalera2
galera:
  mariabackup:
    password: password
```
Doesn't seem to work. The initial pod seems to go up, but then it errors and hits CrashLoopBackOff.
```
Events:
  Type     Reason            Age                From               Message
  ----     ------            ----               ----               -------
  Warning  FailedScheduling  59m                default-scheduler  0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims.
  Normal   Scheduled         59m                default-scheduler  Successfully assigned default/lanegalera2-mariadb-galera-0 to minikube
  Normal   Created           56m (x4 over 59m)  kubelet            Created container mariadb-galera
  Normal   Started           56m (x4 over 59m)  kubelet            Started container mariadb-galera
  Warning  Unhealthy         56m (x3 over 58m)  kubelet            Readiness probe failed: mysqladmin: connect to server at 'localhost' failed
error: 'Can't connect to local server through socket '/opt/bitnami/mariadb/tmp/mysql.sock' (2)'
Check that mariadbd is running and that the socket: '/opt/bitnami/mariadb/tmp/mysql.sock' exists!
  Normal   Pulled   44m (x8 over 59m)      kubelet  Container image "docker.io/bitnami/mariadb-galera:10.7" already present on machine
  Warning  BackOff  3m56s (x222 over 58m)  kubelet  Back-off restarting failed container

```
Meanwhile, I created another install using this config to check that what I saw earlier this morning would happen again:
```yaml
image.tag: 10.7
rootUser.password: password
db.user: testingtwo
db.password: password
db.name: testingtwo
galera.mariabackup.password: password
```
And it did happen again. Since the database is named `my_database` and the image is a 10.6 image. 

From describe:
```
 Environment:
      MY_POD_NAME:                          testtwo-mariadb-galera-0 (v1:metadata.name)
      BITNAMI_DEBUG:                        false
      MARIADB_DEFAULT_PORT_NUMBER:          3306
      MARIADB_GALERA_CLUSTER_NAME:          galera
      MARIADB_GALERA_CLUSTER_ADDRESS:       gcomm://testtwo-mariadb-galera-0.testtwo-mariadb-galera-headless.default.svc.cluster.local,testtwo-mariadb-galera-1.testtwo-mariadb-galera-headless.default.svc.cluster.local,testtwo-mariadb-galera-2.testtwo-mariadb-galera-headless.default.svc.cluster.local
      MARIADB_ROOT_USER:                    root
      MARIADB_ROOT_PASSWORD:                <set to the key 'mariadb-root-password' in secret 'testtwo-mariadb-galera'>  Optional: false
      MARIADB_DATABASE:                     my_database
      MARIADB_GALERA_MARIABACKUP_USER:      mariabackup
      MARIADB_GALERA_MARIABACKUP_PASSWORD:  <set to the key 'mariadb-galera-mariabackup-password' in secret 'testtwo-mariadb-galera'>  Optional: false
      MARIADB_ENABLE_LDAP:                  no
      MARIADB_ENABLE_TLS:                   no

# ....
Normal   Pulled            57m   kubelet            Container image "docker.io/bitnami/mariadb-galera:10.6.10-debian-11-r0" already present on machine
```
